ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py

```diff
@@ -36,7 +36,8 @@ def test_definition(self) -> dict | None:
 
         for idx, interface in enumerate(ethernet_interfaces):
             self.update_interface_shutdown(interface)
-            if not self.validate_data(data=interface, data_path=f"ethernet_interfaces.[{idx}]", required_keys=required_keys, type="routed", shutdown=False):
+            interface_settings = {"switchport.enabled": False, "shutdown": False}
+            if not self.validate_data(data=interface, data_path=f"ethernet_interfaces.[{idx}]", required_keys=required_keys, **interface_settings):
                 continue
```

Comment:
I think this will only work for the new model. If someone was feeding in manually created structured config it would not pick up the old `type: routed`. I think we need to an `not <...> and not <...>` with the old and the new validate_data.

Reply:
or maybe we just check on the presence on an IP address if that is what we need in the test.

Reply:
reading the whole logic here I think I would just check for the shutdown: False and ignore the switchport setting completely. If we have an IP address I think we can assume it is a routed interface.

Reply:
Fixed